### DISCUSSION TITLE
[Deployment] File .env not using

### DISCUSSION THREAD

**Discussion by bynicodevelop** [nuxt/nuxt#18011] [Questions]
### Environment

Nuxt CLI v3.0.0-27376943.f219f63                                                                                                        14:39:55
RootDir: /Users/nickleus/Documents/projects.nosync/com.atrap.io/project                                                            14:39:57
Nuxt project info:                                                                                                                      14:39:57

------------------------------
- Operating System: `Darwin`
- Node Version:     `v16.12.0`
- Nuxt Version:     `3.0.0-27376943.f219f63`
- Package Manager:  `yarn@1.22.10`
- Bundler:          `Vite`
- User Config:      `srcDir`, `css`, `build`, `privateRuntimeConfig`, `vite`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

Hello,

I am testing the deployment on Firebase with nuxt3 in SSR

At the root of the project I have an .env file with a value in it.

```
ENV_VARIABLE=string
```

In nuxt.config.ts

```ts
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    privateRuntimeConfig: {
      VARIABLE: process.env.ENV_VARIABLE
    },
})
```

To execute the Deployment on the hosting of the emulator, I have the command local : `yarn local`

```json

  "private": true,
  "scripts": {
    "build": "nuxi build",
    "clean": "rm -rf .nuxt/ .output/ node_modules/",
    "preset": "NITRO_PRESET=firebase yarn build", <-- As stated in the documentation
    "local": "yarn clean && yarn && yarn preset && firebase emulators:start",
  },
  "devDependencies": {
    "nuxt3": "latest"
  },
}
```

I use this variable in nuxtjs api

A folder /server/api/folder/file.js

with the following code to test:

```ts
export default async (req, res) => {
  return process.env;
}
```

### Describe the bug

The VARIABLE environment variable is not present in the list.

### Logs

```shell
Aucun log
```

**Comment by danielroe** [comment]
It's important the runtime config and env variable names match.

```js
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
    privateRuntimeConfig: {
      ENV_VARIABLE: process.env.ENV_VARIABLE
    },
})
```

See https://github.com/nuxt/framework/pull/2819.

**Comment by bynicodevelop** [comment]
Thank-you for your prompt response.

But I already tried this solution and just tried it again.

It does not work.

In the server function, the variable is considered as undefined.

For on the development server, I manage to access it fine.
Even with different variable names

**Reply by danielroe** [reply]
In your server, you should access it [like this](https://v3.nuxtjs.org/docs/usage/runtime-config#api-routes).

**Reply by bynicodevelop** [reply]
Oh !

I keep wanting to use process. Nice! Thanks 🙏
